well-seasoned-app/src/components/BoardDropdown.js

```diff
@@ -0,0 +1,82 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { useNavigate, useLocation } from "react-router-dom";
+import "./BoardDropdown.css";
+import "../App.css";
+
+const BoardDropdown = ({ boardData, chooseBoard, chosenBoard }) => {
+  const navigate = useNavigate();
+  const location = useLocation();
+  const dropdownOptions = boardData.map((board) => {
+    if (board.title === chosenBoard.title) {
+      return (
+        <option
+          key={board.board_id}
+          id={board.board_id}
+          value={JSON.stringify(board)}
+          selected
+        >
+          {board.title}
+        </option>
+      );
+    } else {
+      return (
+        <option
+          key={board.board_id}
+          id={board.board_id}
+          value={JSON.stringify(board)}
+        >
+          {board.title}
+        </option>
+      );
+    }
+  });
```

Comment:
We can use the spread operator to pass a dictionary of values to an element as props. This allows us to reduce duplication when we have a prop we want to set conditionally.

```javascript
// Set the data that all option elements will have
const optionProps = { 
    key: board.board_id,
    id: board.board_id, 
    value: JSON.stringify(board)
}
    
// Conditionally add `selected` to the optionProps if chosenBoard is found
if (board.title === chosenBoard.title) {
    optionProps["selected"] = true
}
    
// return an option element with the props defined in optionProps
return (<option {...optionProps}>{board.title}</option>);
```